azurerm/resource_arm_log_profile.go

```diff
@@ -0,0 +1,264 @@
+package azurerm
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/2018-03-01/insights"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmLogProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmLogProfileCreateOrUpdate,
+		Read:   resourceArmLogProfileRead,
+		Update: resourceArmLogProfileCreateOrUpdate,
+		Delete: resourceArmLogProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"storage_account_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: azure.ValidateResourceIDOrEmpty,
+			},
+			"service_bus_rule_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: azure.ValidateResourceIDOrEmpty,
+			},
+			"locations": {
+				Type:     schema.TypeList,
+				MinItems: 1,
+				Required: true,
+				Elem: &schema.Schema{
+					Type:             schema.TypeString,
+					StateFunc:        azureRMNormalizeLocation,
+					DiffSuppressFunc: azureRMSuppressLocationDiff,
+				},
+			},
+			"categories": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Schema{
+					Type:             schema.TypeString,
+					DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				},
+			},
+			"retention_policy": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"days": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  0,
```

Comment:
Yup, if the `retention_policy.enabled` is set to true then a value of 0 will retain the events indefinitely. Otherwise events will be kept for the number of days specified.

If the `retention_policy.enabled` is set to false then the value doesn't really matter but the SDK still expects an int to be passed. The default was intended for when retention policy is not enabled.

Ok to remove the default, but would need to mark the days field as required, not optional. Thoughts? @tombuildsstuff